### PR TITLE
[10.0][ADD] mail_parent_recipient

### PR DESCRIPTION
**Pull request by TDu**
This is an attempt to resolve the comments on #329

### CODE REVIEW

File: mail_parent_recipient/tests/test_mail_parent_recipient.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.addons.mail.tests.common import TestMail
+
+
+class TestMailTemplate(TestMail):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestMailTemplate, cls).setUpClass()
```
Comment (simahawk) [inline mail_parent_recipient/tests/test_mail_parent_recipient.py:12]: tracking_disable=True at least or creating these partners :pray:

File: mail_parent_recipient/models/mail_template.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        Walk up the hierarchy of recipient partners via `parent_id`
+        and pick the 1st one having an email.
+        """
+        results = super(MailTemplate, self).generate_recipients(
+            results,
+            res_ids
+        )
+        test_condition = (
+            config['test_enable'] and not self.env.context.get(
+                'test_parent_mail_recipient'
```
Comment (simahawk) [inline mail_parent_recipient/models/mail_template.py]: why do we need this flag?

**Review by simahawk — CHANGES_REQUESTED** [review]

File: mail_parent_recipient/models/mail_template.py (hunk 2 above)
Comment (TDu) [reply mail_parent_recipient/models/mail_template.py]: It is to enable the machinery of this module during test ONLY when the 'test_parent_mail_recipient'  is set.

Comment (simahawk) [reply mail_parent_recipient/models/mail_template.py]: This is not answering the main question: why do we need this flag?
I guess is to not make other tests fail.
IMO this should be configured either at company level (off by default) or at partner level (off by default).
We can start adding a flag at company level 1st + res.config.settings then we'll see.

**Comment by simahawk** [comment]
I would try to move to GH actions on 10.0 branch first.

**Comment by simahawk** [comment]
@TDu please move the update dotfiles to another PR

**Comment by simahawk** [comment]
#972 is getting merged. You can rebase soon :tada:

File: mail_parent_recipient/readme/CONTRIBUTORS.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+* Thomas Nowicki <thomas.nowicki@camptocamp.com>
```
Comment (simahawk) [inline mail_parent_recipient/readme/CONTRIBUTORS.rst:1]: @TDu you can add your name here

File: mail_parent_recipient/models/company.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+
+    _inherit = "res.company"
+
+    use_parent_mail_address = fields.Boolean(
+        string="Use Parent Mail Address",
+        help=""
```
Comment (simahawk) [inline mail_parent_recipient/models/company.py]:
```suggestion
        help="When an email is sent, fallback to partner's parent email if no email is set on the recipient partner."
```

**Comment by simahawk** [comment]
we'll maintain only from v15

**Pull request closed without merging**